src/gui/qgsmaptooldigitizefeature.cpp

```diff
@@ -92,6 +92,40 @@ void QgsMapToolDigitizeFeature::setCheckGeometryType( bool checkGeometryType )
   mCheckGeometryType = checkGeometryType;
 }
 
+void QgsMapToolDigitizeFeature::keyPressEvent( QKeyEvent *e )
+{
+  QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( mLayer );
```

Comment:
This whole block should be inside the `e->key() == Qt::Key_C` condition, and if the key isn't 'c' then you need to delegate to QgsMapToolCapture::keyPressEvent (or pressing escape/backspace will be broken)

Reply:
Right. Done.